### ISSUE TITLE
Unexpected nullable warning from default in switch expression

### ISSUE THREAD

**Issue by cston** [dotnet/roslyn#39888]
The following should compile without warnings:
```C#
#nullable enable

using System.Diagnostics.CodeAnalysis;

class Program
{
    [return: MaybeNull] static T F<T>(bool b, T t)
    {
        return b switch
            {
                false => t,
                _ => default // warning CS8601: Possible null reference assignment
            };
    }
}
```
